torchdata/datapipes/iter/__init__.pyi.in

```diff
@@ -0,0 +1,100 @@
+${init_base}
+
+########################################################################################################################
+# The part below is generated by parsing through the Python files where IterDataPipes are defined.
+# This base template ("__init__.pyi.in") is generated from mypy stubgen with minimal editing for code injection
+# The output file will be "__init__.pyi".
+# Note that, for mypy, .pyi file takes precedent over .py file, such that we must define the interface for other
+# classes/objects here, even though we are not injecting extra code into them at the moment.
+
+from .util.extractor import CompressionType
+from torchdata.datapipes.map import MapDataPipe
+from torch.utils.data import DataChunk, IterableDataset
+from torch.utils.data._typing import _DataPipeMeta
+
+from typing import Any, Callable, Dict, List, Optional, Sequence, Tuple, TypeVar, Union
+
+try:
+    import torcharrow
+except ImportError:
+    torcharrow = None
+
+T_co = TypeVar("T_co", covariant=True)
+
+class IterDataPipe(IterableDataset[T_co], metaclass=_DataPipeMeta):
+    functions: Dict[str, Callable] = ...
+    reduce_ex_hook: Optional[Callable] = ...
+    getstate_hook: Optional[Callable] = ...
+    def __getattr__(self, attribute_name: Any): ...
+    @classmethod
+    def register_function(cls, function_name: Any, function: Any) -> None: ...
+    @classmethod
+    def register_datapipe_as_function(
+        cls, function_name: Any, cls_to_register: Any, enable_df_api_tracing: bool = ...
+    ): ...
+    def __getstate__(self): ...
+    def __reduce_ex__(self, *args: Any, **kwargs: Any): ...
+    @classmethod
+    def set_getstate_hook(cls, hook_fn: Any) -> None: ...
+    @classmethod
+    def set_reduce_ex_hook(cls, hook_fn: Any) -> None: ...
+    # Functional form of 'BatcherIterDataPipe'
+    def batch(self, batch_size: int, drop_last: bool = False, wrapper_class=DataChunk) -> IterDataPipe: ...
```

Comment:
Why do we need to specify all functional APIs from PyTorch Core here?
Why not to use `gen_pyi` to generate these APIs as torchdata depends on torch?

Reply:
Good catch - I have updated both PRs to make it possible to do that.